### ISSUE TITLE
[TF2] Shadows seen through the wall

### ISSUE THREAD

**Issue by jorgenpt** [ValveSoftware/Source-1-Games#52]
Issue transferred from ValveSoftware/steam-for-linux#420
@DarthBrandon posted at 2012-12-27T01:12:19Z:

![2012-12-26_00003](https://f.cloud.github.com/assets/3124075/31922/463dc728-4fc2-11e2-80ce-ddca560f4e81.jpg)
During setup, I'm seeing the shadows of BLU team members through a wall of their base.

Processor Information:
    Vendor:  GenuineIntel
    Speed: 2801 Mhz
    4 logical processors
    4 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported

Network Information:
    Network Speed:  

Operating System Version:
    Ubuntu 12.04.1 LTS (64 bit)
    Kernel Name:  Linux
    Kernel Version:  3.2.0-35-generic
    X Server vendor:  The X.Org Foundation
    X Server release:  11103000

Video Card:
    Driver:  NVIDIA Corporation GeForce GT 240/PCIe/SSE2

```
Driver Version:  3.3.0 NVIDIA 310.14
Desktop Color Depth: 24 bits per pixel
Monitor Refresh Rate: 60 Hz
VendorID:  0x10de
DeviceID:  0xca3
Number of Monitors:  1
Number of Logical Video Cards:  1
Primary Display Resolution:  1280 x 1024
Desktop Resolution: 1280 x 1024
Primary Display Size: 14.80" x 11.85"  (18.94" diag)
                                        37.6cm x 30.1cm  (48.1cm diag)
Primary Bus: PCI Express 16x
Primary VRAM: 1024 MB
Supported MSAA Modes:  2x 4x 8x 16x 
```

Sound card:
    Audio device: Realtek ALC888

Memory:
    RAM:  5927 Mb

Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Microphone:  Not set
    Total Hard Disk Space Available:  187739 Mb
    Largest Free Hard Disk Block:  23387 Mb

Installed software:

Recent Failure Reports:

**Comment by kisak-valve** [comment]
Hello, can someone confirm if this issue is still occurring with up to date video drivers?

**Comment by hahagoni** [comment]
Yes, the issue persists with up-to-date drivers
Radeon Software Version - 16.8.2
Radeon Software Edition - Crimson
Graphics Chipset - AMD Radeon R9 200 Series
Memory Size - 2048 MB
Memory Type - GDDR5
Core Clock - 1070 MHz
Windows Version - Windows 8.1 (64 bit)
System Memory - 8 GB
CPU Type - Intel(R) Core(TM) i3-4330 CPU @ 3.50GHz
Radeon Settings Version - 2016.0916.1515.27418
Driver Packaging Version - 16.30.2511.1001-160811a-305723E
Provider - Advanced Micro Devices, Inc.
2D Driver Version - 8.01.01.1564
Direct3D® Version - 9.14.10.01209
OpenGL® Version - 6.14.10.13447
OpenCL™ Version - 6.14.10.13447
AMD Mantle Version - 9.1.10.0136
AMD Mantle API Version - 98309
AMD Audio Driver Version - 9.0.0.9910
Vulkan™ Driver Version - 1.2.0
Vulkan™ API Version - 1.0.17

**Comment by covertoperator** [comment]
Yes, this is a very common bug that everyone has learned to live with. It happens whenever there is a thin wall or ceiling (not sure about floors) that a player standing right next to (i.e. they have walked into it)

 For example, look at the sloped roof underneath the lower platform in the centre of hightower (above the lower health packs room). If there is a player standing on the sloped roof, you can see their shadow on the floor above them. I may create and upload screenshots on request.

**Comment by kamild1996** [comment]
I'm on Crimson ReLive (16.12.1), still happens. Actually, I don't remember since when it's happening, it was so long...

**Comment by hahagoni** [comment]
I think it's a bug occuring in other source games aswell

**Comment by henke37** [comment]
That last video seems to be for fancier stuff like dynamic lights and projected textures. This issue is just about plain old dynamic shadows that entities do with normal static lights.

**Comment by thejaviertc** [comment]
This bug still exists, has been in the game for a long time and is still happening to this day.

It can be used to know that someone is behind a corner or wall, like this video that I found on Reddit.

https://user-images.githubusercontent.com/45662206/178162138-ffef17ed-251a-41e6-aed4-275493fc79b5.mp4

**Comment by kamild1996** [comment]
What a blast from the past :) But yes, shadows still broken

**Comment by Ashetf2** [comment]
I just wanted to mention that toolsblocklight doesn't work to prevent this:

![image](https://user-images.githubusercontent.com/63692647/204612790-da7209fa-cffc-4a5d-b364-19c92f296057.png)

![image](https://user-images.githubusercontent.com/63692647/204612932-bee8784a-fe84-4c2e-97a7-519774ee5365.png)

**Comment by stevethepocket** [comment]
Of course it doesn't. Light entities have nothing to do with dynamic shadow generation.

**Comment by Ashetf2** [comment]
@kisak-valve you may want to remove the "Future" milestone? It's been 9 years and there are no signs of this being fixed soon.

Also, if Valve or Joshua decides to fix this, I suggest to implement dynamic RTT shadow angles. There is already a tutorial on how to do it in the VDC: https://developer.valvesoftware.com/wiki/Dynamic_RTT_shadow_angles_in_Source_2007. Right now dynamic shadows uses light_environment as the source, and it looks ugly in interiors.

**Comment by Ashetf2** [comment]
I believe Valve when making CS:GO just gave up trying to fix this bug and then moved to CSM shadows. So, maybe for the summer update, instead of fixing this bug (if possible), we could get CSM shadows. This is an example on how it would look like.

(This is a modification to the game, it isn't CS:GO)
![image](https://user-images.githubusercontent.com/63692647/220751217-6086a820-33fe-464b-b531-74e478ada180.png)

Of course, this would cause a drop in fps for low-end machines, so if it is implemented, RTT shadows may have to be kept but with player shadows disabled.
